nifi-docs/src/main/asciidoc/administration-guide.adoc

```diff
@@ -164,6 +164,16 @@ accomplished by setting the `nifi.remote.input.secure` and `nifi.cluster.protoco
 
 In order to facilitate the secure setup of NiFi, you can use the `tls-toolkit` command line utility to automatically generate the required keystores, truststore, and relevant configuration files. This is especially useful for securing multiple NiFi nodes, which can be a tedious and error-prone process.
 
+Wildcard certificates (i.e. two nodes `node1.nifi.apache.org` and `node2.nifi.apache.org` being assigned the same certificate with a CN or SAN entry of +*.nifi.apache.org+) are *not officially supported* and *not recommended*. There are numerous disadvantages to using wildcard certificates, and a cluster working with wildcard certificates has occurred in previous versions out of lucky accidents, not intentional support.
```

Comment:
@alopresto - I agree with you that wildcard certificates should be avoided, but I just want to clarify something for my own curiosity. Let's say we have:
- Node 1 (certificate with CN=`node1.nifi.apache.org`, SAN=`*.nifi.apache.org`)
- Node 2 (certificate with CN=`node2.nifi.apache.org`, SAN=`*.nifi.apache.org`)

That would not cause any issue regarding how NiFi clustering is working, right? Because we would have a unique DN for every node. I imagine people could have such certificates because, for instance, they don't know in advance what would be the hostname of a load balancer in front of NiFi.

Having said that, I agree that it's a security concern in case a certificate is compromised.

Reply:
Having a SAN entry with a wildcard value is perfectly acceptable *IF* there is another SAN with a unique value and the DN is unique. Older advice would have just been to provide a unique DN, but with the way Google Chrome and other tools are changing to come inline with [RFC 6125](https://tools.ietf.org/html/rfc6125#section-6.4.4), DN is no longer a preferred definitive provider for the CN. It is supposed to be used only as a last resort if SAN is missing. I have filed a Jira [NIFI-5398](https://issues.apache.org/jira/browse/NIFI-5398) to improve NiFi's handling of this situation as well.